src/dunst.c

```diff
@@ -72,12 +72,12 @@ void check_timeouts(void)
                         continue;
                 }
 
+                /* shift iter before we possibly delete the current notification */
+                iter = iter->next ? iter->next : NULL;
```

Comment:
Even if it's meant to be `iter` it's the same thing, if `iter` is NULL it wouldn't have got past the while loop check so it wouldn't reach this line.